packages/react-instantsearch-hooks-dom/src/ui/Pagination.tsx

```diff
@@ -52,26 +96,34 @@ export function Pagination({
           <PaginationItem
             isDisabled={isFirstPage}
             className="ais-Pagination-item--firstPage"
-            aria-label="First"
+            aria-label={translations.ariaFirst}
             href={createURL(firstPageIndex)}
             onClick={() => onNavigate(firstPageIndex)}
           >
-            ‹‹
+            {translations.first}
           </PaginationItem>
         )}
         {showPrevious && (
           <PaginationItem
             isDisabled={isFirstPage}
             className="ais-Pagination-item--previousPage"
-            aria-label="Previous"
+            aria-label={translations.ariaPrevious}
             href={createURL(previousPageIndex)}
             onClick={() => onNavigate(previousPageIndex)}
           >
-            ‹
+            {translations.previous}
           </PaginationItem>
         )}
         {pages.map((page) => {
-          const label = String(page + 1);
+          const label =
+            typeof translations.page === 'string'
+              ? translations.page
+              : translations.page(page);
```

Comment:
We should rather check for `typeof x === 'function'` and execute it if so, instead of the opposite, because here you'll easily end up calling a `number` or any other data types.

Reply:
We probably want a function all the time here though, similar to https://github.com/algolia/react-instantsearch/pull/3387#discussion_r833014745.

Reply:
Addressed in 91703928.